### ISSUE TITLE
Running ADS-WS1 getting error 'unexpected buffer length'

### ISSUE THREAD

**Issue by jezkerwin** [weewx/weewx#372]
I'm running a Argent Data ADS-WS1 weather station on a raspberry Pi.

When looking at /var/log/messages, I see the following errors and then after a while the engine shuts down.
Occasionally a successful reading happens, but for the most post it's these messages.

`Feb 11 21:11:04 weewx[6897]: ws1: Failed attempt 1 of 5 to get readings: Unexpected buffer length 51
Feb 11 21:11:31 weewx[6897]: ws1: Failed attempt 2 of 5 to get readings: Unexpected buffer length 594
Feb 11 21:12:11 weewx[6897]: ws1: Failed attempt 1 of 5 to get readings: Unexpected buffer length 730
Feb 11 21:12:21 weewx[6897]: ws1: Failed attempt 2 of 5 to get readings: Unexpected buffer length 102
`

Does anyone know what this means, and how to fix?
Thanks.

**Comment by jezkerwin** [comment]
Here is an excerpt from /var/log/syslog

```
Feb 11 21:18:09 . weewx[7001]: restx: AWEKAS: Posting not enabled.
Feb 11 21:18:09 . weewx[7001]: engine: Finished loading service weewx.restx.StdAWEKAS
Feb 11 21:18:09 . weewx[7001]: engine: Loading service weewx.engine.StdPrint
Feb 11 21:18:09 . weewx[7001]: engine: Finished loading service weewx.engine.StdPrint
Feb 11 21:18:09 . weewx[7001]: engine: Loading service weewx.engine.StdReport
Feb 11 21:18:09 . weewx[7001]: engine: Finished loading service weewx.engine.StdReport
Feb 11 21:18:09 . weewx[7001]: engine: Starting up weewx version 3.9.1
Feb 11 21:18:09 . weewx[7001]: engine: Station does not support reading the time
Feb 11 21:18:09 . weewx[7001]: engine: Starting main packet loop.
Feb 11 21:18:09 . weewx[7001]: manager: Daily summary version is 2.0
Feb 11 21:18:38 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 33 34 37 30 33 34 38 30 33 34 38 30 30 34 34 34 30 38 33 30 37 30 33 37 3C 36 3C 0D EA 21 21 30 30 30 37 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 42 3C 33 34 37 30 33 34 38 30 33 36 3C 30 3C 3B 37 36 3C 36 3B 37 37 38 B3 3F 3C 37 39 ED DB 21 21 30 30 30 37 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 61 30 33 34 37 38 33 34 3C 30 33 37 3E 34 3C 37 37 37 34 B6 31 3E 36 3C B3 FB FB 8A 6A 4F 21 21 30 30 30 39 30 30 46 44 30 30 30 38 38 30 30 36 32 32 41 72 36 33 37 37 3C 33 37 3B 34 BB B7 39 32 36 BF FA 9A FA 87 FA BB AA DA FF FF FF EA 7B FD 21 21 30 30 30 39 30 30 46 44 30 30 38 30 38 30 30 36 3A 3A 61 52 3B 33 37 37 36 33 53 B2 DA BB FB 92 BA FE FA 9A B2 FA FF FF FF DA DA BF FB EA EB 21 21 30 30 30 43 30 30 46 44 30 30 3A 32 36 38 34 34 3B 3B 71 7A 3F 3B 36 37 B7 3B 4D E2 DA BB EB 92 BA FE E7 E7 AB DA FF FF F3 9B DB FB FF 9A FE 21 21 30 30 30 43 30 30 46 64 30 30 37 34 3B 34 3C 9B A7 A7 E9 F5 B7 D3 D3 9F AB EF C2 DA B3 FF AF BA FE FF FF A3 FA FF FF FB DF F6 FF 21 21 30 30 30 43 30 30 46 44 30 30 3E 38 3A 34 3C 36 3B 3A 59 52 DF BB BA BA 9B BA FB C2 DE A7 FF CF BA FB FB FB AB FA FF 3F FF DF FB 21 21 30 30 30 45 30 3C 66 46 32 34 3B 37 3C 32 3C DB D6 8B E9 BF B7 DB D3 DF FF CF FA DE F3 E3 8A FF FF BF FA FF DA D3 FF 21 21 30 38 38 45 30 34 66 46 33 3C 3E 33 39 37 3E FB DF 6B ED 3F 76 D3 93 BF FB FA DA 9F EA FF FF DF FF 21 21 30 34 31 32 30 38 47 77 37 3C BF 4D CB D2 BB DB DB EB CD DF FF BB F6 BF FF F2 DF FF FF FF FE FF 21 21 30 34 34 45 30 3A 66 45 37 3C 3F DF 9D 9F B6 D7 DE 8B C9 DF FB BB EE E7 FF FB DE F7 FF FB 21 21 30 3C 31 36 38 3E 67 47 35 3C BF BB BB 3B 36 DF DF EA ED DB BF BB FA FF FF FF FE FF FE FF FF FF 21 21 30 38 36 65 38 3E 47 77 3E 3C BF F6 BF F2 FE DF DB EB ED FA F3 BB DE FF BF DE FB FF BF 21 21 30 3B 39 3B 38 B8 67 77 3F 36 BF DF FF BA BA FB DF CF ED FF 73 BB DB BF 21 21 38 3F 3D 3F 34 3A F7 F7 BA FA BB F3 FB BA FB FF FF 9F F5 7B FF DF FF 21 21 30 3E 31 3C 37 38 DF FF 3B BF FF FB AB F7 EB D3 FF DF 21 31 38 3C 4F BA B2 BE 77 AB 8A FF FF FF DF FE 4B FF EE FF F7 21 39 34 3E EF 9B B2 B2 CF BF FA FF BF 5F 21 31 30 38 6F DB BA B2 FF FA F6 BB FF'
Feb 11 21:18:38 . weewx[7001]: ws1: Failed attempt 1 of 5 to get readings: Unexpected buffer length 730
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 31 35 30 30 46 43 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 41 0D 0A'
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 31 33 30 30 46 43 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 42 0D 0A'
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 31 33 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 42 0D 0A'
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 31 33 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 42 0D 0A'
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 31 33 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 43 0D 0A'
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 31 30 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 43 0D 0A'
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 31 30 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 41 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 43 0D 0A'
Feb 11 21:18:48 . weewx[7001]: ws1: bytes: '21 21 30 30 30 45 30 30 46 44 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 38 38 30 43 0D 0A'
Feb 11 21:19:18 . weewx[7001]: ws1: bytes: '21 21 30 30 30 45 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 41 30 33 34 37 30 33 34 38 30 33 34 38 30 30 34 30 3C 30 38 36 36 36 30 33 34 36 34 43 8D CA 21 21 30 30 30 45 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 36 36 36 38 38 34 39 3A 30 33 3B 3B 37 43 ED EB 21 21 30 30 30 45 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 41 36 33 34 37 30 33 36 3A 30 33 35 3C 32 38 3C 37 37 33 34 3F 3A 3A 37 B3 FB F3 3A 35 AB 21 21 30 30 30 43 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 3C 33 34 37 30 33 36 3C 3C 33 34 3F 36 34 3E 3B 37 3B 36 BF B3 3F 99 A6 FB D2 23 35 EF 21 21 30 30 30 43 30 30 46 44 30 30 30 38 3C 3C 30 34 3A 3A 51 42 3C 33 37 37 30 33 B7 3D 37 33 B7 3F 3D 34 3E BF DF 9A DA FE FB FF 9A F2 FF FB 7F B5 FF 21 21 30 30 30 43 30 30 46 43 30 38 30 38 34 36 36 34 32 3A 7D 62 34 BB 37 37 3A 33 B7 3F 3E 33 B6 DA FA 9A B2 FF C7 FB C2 FA F3 FF 9F FA F7 FF CF FB 21 21 30 30 30 39 30 30 46 42 34 3C 30 38 3B 3E 3A 34 32 3B 7D 5B 38 B3 DB D3 9A B2 DB DB 9F B2 FF FF FF 87 FA FB FB FB DA FA FF FB FE 4F FF 21 21 30 30 30 43 30 38 46 62 3E 34 36 38 38 35 37 35 32 B7 7D 5B 34 B3 DB 5B 9A B2 FF FF DF BB FA FB FB F3 C2 FF FF EA FF FF FF EB 21 21 30 30 30 39 30 34 46 42 36 3A 36 38 3C 3B 3D 37 32 BB 57 57 9A FA BF D6 9F BA FF EF DE AB FA FF FF FF FB FA FF F7 FB FF FF 21 21 30 38 30 43 34 32 46 62 3A 37 3A 33 3A 3F B6 DA DA EB D5 F9 33 B5 FF DF BB CF FF 3F F3 F6 F7 FF FF FF 21 21 38 38 38 39 34 3A 66 62 30 3F DD A2 BA F3 BB 9B DA EA FB BF B6 FB FF DB BB F3 FF FF FF FA FF FB FF 21 21 38 3C 30 39 38 33 46 5B 34 BE 3F 9D B2 FA B7 97 9E EA FF BF B7 D3 FB DE F6 FE FE FF AF FB FB FF 21 21 38 34 36 39 38 B2 BB 2D C2 FA FB FF BE FA FF FF FF EA ED DB FB BB FE D7 FF FB FE FF 21 21 30 36 3B 3D 34 3C 77 7B 3F 36 BF FF FF BB EE FF FF EE D5 FF B3 DF DF FE 21 31 34 34 38 37 39 32 F7 3B F7 DA FA FB FB BF BA FF FD DA F7 21 31 32 30 BE 37 3F 37 46 BD FB B3 BA FE FF FF BF D7 FF 9F FF FF F6 FF 21 39 3C 3B 3C B7 B7 3B F5 BF FF B7 FA FF B7 F7 21 B9 3F 37 33 34 BB BB E9 ED FF FB D7 FF 9B FF FF FF 21 21 33 35 3F 37 BB BF 7B 3F BB FF EF FF FF 31 39 BD B7 BF BA FE F7 B7 FF FF FF'
Feb 11 21:19:18 . weewx[7001]: ws1: Failed attempt 1 of 5 to get readings: Unexpected buffer length 708
Feb 11 21:19:28 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 38 43 0D 0A'
Feb 11 21:19:28 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 43 0D 0A'
Feb 11 21:19:28 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 42 0D 0A'
Feb 11 21:19:28 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 43 0D 0A'
Feb 11 21:19:28 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 41 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 42 0D 0A'
Feb 11 21:19:28 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 41 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 42 0D 0A'
Feb 11 21:19:28 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 41 0D 0A'
Feb 11 21:19:30 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 38 30 41 0D 8A 21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 38 38 30 30 3C 34 34 33 30 32 32 61 0D 6A 21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 38 38 30 3C 36 36 3C 38 36 3A 34 33 38 35 3A 55 0D 6A 21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 41 30 33 34 37 30 33 34 3C 34 33 34 3E 3A 32 38 34 3B 3E 36 30 3C 3F 3B 33 3C B6 B9 3D 8D 0A'
Feb 11 21:19:30 . weewx[7001]: ws1: Failed attempt 1 of 5 to get readings: Unexpected buffer length 207
Feb 11 21:19:55 . weewx[7001]: ws1: bytes: '21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 61 62 30 33 34 37 3C 33 34 3E 36 33 34 3C 3F 3B 33 34 3F 3B 3F 32 32 BB 9A BA B2 FA FB 6A CA 21 21 30 30 30 32 30 30 30 30 30 30 30 30 30 30 30 34 32 32 41 51 3C 33 34 37 3C 33 34 3C 3A 33 36 38 38 3D 38 36 36 9F FA 82 F2 FA DB FB F2 F2 CF 6F 8B 21 21 30 30 30 30 30 30 30 30 30 36 34 32 30 3A 3B 34 32 32 7D 4B 3F 33 B4 93 DA BA FA F7 DE BE FA FA FF BF BF FA FA F3 FB D2 F2 FF F7 F6 BF FF 21 21 30 30 30 30 38 3C 34 30 30 3F 3B 3F 34 36 4F DB D2 EA 7F 9E BA BA BA DF BE E2 FE DF EF F7 DA FF FF FF EB E3 FF FF FB FB FF 21 21 30 38 38 30 38 36 36 33 3C 3F 3D 3F 36 3E B7 D7 9B EA E9 FD B7 D3 FB DF FF FB FE B7 FF DA FF F7 FF FF 21 21 38 34 34 30 34 3D 37 3C 36 3E 97 F6 A2 FA BB DB DA EA F5 DF DB DB BF FB 9A FF FF FF F7 FF 21 21 38 32 34 32 38 3E 3F 37 31 36 CF EB A2 FA FE FF DF AB DD BF BB EB FE FF EB FF FF FF FF 21 21 30 37 38 36 32 B6 A7 F6 A2 FA BA D7 DB FA F6 FF FF 5F 7D FD F7 BF CB FF DF FF FF 21 21 30 3E 3D 37 37 34 BF B7 9F A6 FF FF FE EF F6 F7 FF FB FF DF FF 21 39 30 34 9D D3 B2 EA DF D7 C7 E2 F6 FF BB B6 FF 6F FF DF F7 21 31 33 38 BB B7 BD 3A 36 B7 DF 97 DA FF DB BF F7 FF 21 29 37 34 BE FE BF AA BA FB FB BB FB FF D7 FF FF A1 B5 33 37 3A FF E7 EF EA FD FB EB FB FF FB F7 21 37 BF 4D FA FB FB EF D3 F3 EE FF FF'
Feb 11 21:19:55 . weewx[7001]: ws1: Failed attempt 2 of 5 to get readings: Unexpected buffer length 429
Feb 11 21:20:05 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 32 46 34 38 30 30 38 3E 32 36 32 3A 4D 6A 36 33 36 37 3A 33 34 3F 3F 33 34 38 3F 3F 3B 34 36 37 3B 35 30 B7 37 BB 37 3A BF B3 8D 4A 21 21 30 30 30 30 30 30 32 46 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 32 0D 0A'
Feb 11 21:20:05 . weewx[7001]: ws1: Failed attempt 3 of 5 to get readings: Unexpected buffer length 102
Feb 11 21:20:15 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 32 46 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 31 0D 0A'
Feb 11 21:20:15 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 32 46 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 31 0D 0A'
Feb 11 21:20:15 . weewx[7001]: manager: Added record 2019-02-11 21:20:00 AEDT (1549880400) to database 'weewx.sdb'
Feb 11 21:20:15 . weewx[7001]: manager: Added record 2019-02-11 21:20:00 AEDT (1549880400) to daily summary in 'weewx.sdb'
Feb 11 21:20:15 . weewx[7001]: reportengine: Running reports for latest time in the database.
Feb 11 21:20:15 . weewx[7001]: reportengine: Running report 'SeasonsReport'
Feb 11 21:20:15 . weewx[7001]: reportengine: Found configuration file /etc/weewx/skins/Seasons/skin.conf for report 'SeasonsReport'
Feb 11 21:20:15 . weewx[7001]: cheetahgenerator: using search list ['weewx.cheetahgenerator.Almanac', 'weewx.cheetahgenerator.Station', 'weewx.cheetahgenerator.Current', 'weewx.cheetahgenerator.Stats', 'weewx.cheetahgenerator.UnitInfo', 'weewx.cheetahgenerator.Extras']
Feb 11 21:20:15 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 32 46 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 31 0D 0A'
Feb 11 21:20:15 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 32 46 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 31 0D 0A'
Feb 11 21:20:15 . weewx[7001]: manager: Daily summary version is 2.0
Feb 11 21:20:15 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 32 46 30 30 30 30 30 30 30 34 32 32 41 42 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 31 0D 0A'
Feb 11 21:20:16 . weewx[7001]: ws1: bytes: '21 21 30 30 30 30 30 30 32 46 30 30 30 30 30 30 30 34 32 32 41 41 30 33 34 37 30 33 34 38 30 33 34 38 30 30 30 30 30 30 30 30 30 30 30 33 30 30 30 31 0D 0A'
Feb 11 21:20:16 . weewx[7001]: restx: Wunderground-PWS: Published record 2019-02-11 21:20:00 AEDT (1549880400)
Feb 11 21:20:19 . weewx[7001]: cheetahgenerator: Generated 8 files for report SeasonsReport in 3.78 seconds
Feb 11 21:20:19 . weewx[7001]: manager: Daily summary version is 2.0
Feb 11 21:20:20 . weewx[7001]: imagegenerator: Generated 14 images for SeasonsReport in 0.89 seconds
Feb 11 21:20:20 . weewx[7001]: copygenerator: copied 5 files to /var/www/html/weewx
Feb 11 21:20:20 . weewx[7001]: reportengine: Report 'SmartphoneReport' not enabled. Skipping.
Feb 11 21:20:20 . weewx[7001]: reportengine: Report 'MobileReport' not enabled. Skipping.
Feb 11 21:20:20 . weewx[7001]: reportengine: Report 'StandardReport' not enabled. Skipping.
Feb 11 21:20:20 . weewx[7001]: reportengine: Report 'FTP' not enabled. Skipping.
Feb 11 21:20:20 . weewx[7001]: reportengine: Report 'RSYNC' not enabled. Skipping.
```

**Comment by windcrusader** [comment]
I've seen similar issues from time to time on my WS1 also interfacing through a raspberry pi model B. 

I had a difficult time getting reliable serial interfacing with the unit. Initially I tried a cheap USB to serial converter and this didn't work at all. Eventually, I succeeded by building my own direct to raspberry pi GPIO UART interface using a MAX232 chip. This was a fun electronics project, but quite a bit of effort to go to for just getting a serial interface going. My connection is now pretty reliable though. I understand people have had more success with the USB serial converters based on the FDTI chip.

I think occasionally you also just get bad loop packets over the serial interface. For me, this seems to occur most often during startup or when trying to alter the unit parameters via the command prompt over serial interface (and being too lazy to shut weewx down during the process).

Have you debugged your serial connection e.g tried a different cable, shorter cable, etc etc?

I think this issue is not a weewx issue though, and more of a hardware issue. I would recommend the issue is closed. Agree?
